### DISCUSSION TITLE
Better options to sync content across locales

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8109] [Ideas]
> Created by: Christian (seelbach@gmx.de) on 2015/03/09 00:49:13 +0000
> Votes at time of UserVoice import: 58

-----------------------------------

Right now, the content of all fields which are marked as translatable are only copied over to the other locales on the very first entry save.

Wouldn't it be great if there was a checkbox right next to each field that triggers another synchronization (of that field) across all locales?

There are lots of cases where you need a field to be translatable, but only make locale specific edits in very rare situations. Being able to trigger another synchronization after making edits to such a field saves you from extensive and error-prone copy and past actions.

I can imagine this being a really complex and complicated feature to implement. So an alternative suggestion would be to just provide an indication whether a field is still in sync or not.

To list some example cases: fields for proper names of things that are almost (!) never translated like city names, fields where you only occasionally list things and include conjunctions like "and", date fields that are usually the same across locales but can differ in special cases like release dates.

**Comment by angrybrad** [comment]
> Posted by Christian (seelbach@gmx.de) on 2015/03/09 00:49:13 +0000
-----------------------------------

Great stuff, Tim. All that is left now is a solution to do this on a per field basis and some kind of indication for what fields are still in sync (updated via JS?)...

**Comment by angrybrad** [comment]
> Posted by Timothy Kelty (timkelty@gmail.com) on 2015/04/02 12:33:05 +0000
-----------------------------------

Not much as far as docs yet, but I just made this: https://github.com/timkelty/craft-localesync

Allows you to sync matching content across locales, with the option to override, and set defaults.

**Comment by angrybrad** [comment]
> Posted by Anonymous (brian@tippingmedia.com) on 2015/04/03 17:38:09 +0000
-----------------------------------

We work with a lot of churches with multiple campuses. Some pages have the same content campus-to-campus (such as About Us) while others are locale-specific (such as Ministry date/times). One way we&#39;ve handled this is creating  templates for &quot;Sub Page Per Campus&quot; and Sub Page All Campus&quot;. These allow us to determine how to handle unique fields for each locale. 

Our solution feels a bit &quot;hacked&quot; and I&#39;m interested to see how Christian&#39;s suggestion might work for our church clients (no pun intended). :)

**Comment by angrybrad** [comment]
> Posted by Petter (petter@feed.no) on 2015/06/10 08:24:47 +0000
-----------------------------------

I have used WordPress Multilingual when working in WordPress, and the function i miss in Craft locales is the possibility to sync a full entry to another locale at any time. Would love to see this function. Ex when you have a multi region website, not just multi language, you could have more then one locale with english copy. We have a customer that have 5 locales with English copy, and they have to copy paste the same content five times every time they add an entry. 

Dont know if this should be a separate request, but since Christians title works for me, i&#39;ll start here :)

**Comment by rogertinch** [comment]
I like the idea of a checkbox for "Synch across Sites" next to a translatable field so you can choose which sites should have totally different data while still having the ability to synch data across other sites.

**Comment by timkelty** [comment]
@tinchalamo have you tried https://github.com/timkelty/craft-localesync? Will probably port to Craft 3 soon.

**Comment by dougstjohn** [comment]
We use a custom twig variable solution for one of our clients. This doesn't sync content between locales, but it does allow for individual entry fields to fallback to en-us if no content is set on the current user's locale.

Content managers fill out en-us entries and other locale entries (ex. en-au, en-nz, etc) can stay empty and then if those locale's entries are ever set, that data will be used instead of en-us.

`{{ withEnUsFallback(entry, 'termsOfUseContent') }}`

It would be nice to have a UI where you could easily compare one entry field across all locales.

**Comment by timkelty** [comment]
Just ported Locale Sync to Craft 3 here: https://github.com/timkelty/craft-site-sync
Should be functional, docs to come.

**Comment by christianruhstaller** [comment]
A similar but different strategy for the sync: https://github.com/Goldinteractive/craft3-sitecopy :)

**Comment by outline4** [comment]
There is no logic to synch content across sites until the first save. 

My guess is you've chosen this path because it's easier to program. But nobody in their right mind would do this because it's best practice. It just doesn't make any sense! 

In my opinion, content should be synched to the original content (the one you've started with), until there's a change in any field of other sites. There should be some sort of indicator in which site relies the original content, and that it's not edited yet, respectively still in synch. 

If this is too complex for now, you could maybe start by only checking the title field:  
if the title field has changed, stop synching the content to the "master" content and mark it "translated" or something. 

Also: As soon as the title in another site is changed you should empty the slug that was prefilled from the other site. 

You can do better. And I believe this should be core.  

P.s. we have 4 official languages in our country. We need to take translations serious ;)

**Comment by brandonkelly** [comment]
Craft is going to start tracking dirty fields in 3.4 (#4149), so this is something we could potentially do with that info. May pair well with a fix for #4642 as well.

**Comment by piotrpog** [comment]
@brandonkelly Does it mean that there will be option to copy one locale contents to another locale (cloning sites)?

**Comment by brandonkelly** [comment]
@piotrpog that’s the idea, yeah.

**Comment by janeschindler** [comment]
Does anyone (@brandonkelly) have an update here? I'd be curious if this "sync translation" feature might still happen. That would be great. I'm sorely missing it.

**Reply by brandonkelly** [reply]
Definitely still something we want to look into.

**Comment by outline4** [comment]
Yes! Please!

**Comment by gregorterrill** [comment]
I've been working on a couple of WP sites lately and despite it being a generally awful experience - one thing I really do miss in Craft is something like classic WPML's options to replace your existing content with content from another language (the two buttons at the bottom here):
![wpml-options](https://user-images.githubusercontent.com/1774747/184353768-d7eeea0b-abb8-496c-b9e0-af2c75aa0665.png)

If this were implemented in Craft, my vote would be for:

1.  Like WPML, a button in the edit entry screen sidebar to copy over ALL the entry's field values from a given site. This could maybe be combined somehow with the Status section since all sites are already listed there? Maybe a "Copy content from" button next to each site.
2. When mousing over a translation icon, the tooltip indicating translation settings would also include the option to copy JUST that field value from a given site, like this:
![copy-dropdown](https://user-images.githubusercontent.com/1774747/184354829-2393aef4-5b16-41b0-9e8e-015bc4e282fd.png)

This would be so useful! Hope you're still looking into it.

**Reply by brandonkelly** [reply]
Love that idea.

**Reply by proimage** [reply]
That would be great for sure. Along with that, though, don't forget a way to bulk-update an entire entry. :)

**Reply by brandonkelly** [reply]
@proimage If you mean like #8132, follow along over there for updates.
